sky/clouds/cloud.py

```diff
@@ -140,6 +140,12 @@ def make_deploy_resources_variables(
         """
         raise NotImplementedError
 
+    @classmethod
+    def get_vcpus_from_instance_type(cls,
+                                     instance_type: str) -> Optional[float]:
+        """Returns the number of virtual CPUs that the instance type offers."""
+        raise NotImplementedError
```

Comment:
Can we change this to 
```
return service_catalog.get_vcpus_from_instance_type(instance_type, clouds=repr(self))
```
then we don't need to duplicate the implementation in child classes

Reply:
I didn't do that because 1) local clusters not have service catalogs, and 2) for code consistency (https://github.com/skypilot-org/skypilot/pull/1076#discussion_r948518068). WDYT?

Reply:
Ah I see.. we should probably change that but maybe in another PR. Implementing in child class is okay in this PR for the sake of consistency